certbot-apache/certbot_apache/tests/centos6_test.py

```diff
@@ -116,10 +117,51 @@ def test_loadmod_rootconf_exists(self):
         mods = [lm for lm in root_loadmods if self.config.parser.loc["default"] in lm]
 
         self.assertEqual(len(mods), 1)
+        # [:-7] removes "/arg[#]" from the path
         self.assertEqual(
             self.config.parser.get_all_args(mods[0][:-7]),
             sslmod_args)
 
+    def test_loadmod_non_duplicate(self):
+        # the modules/mod_ssl.so exists in ssl.conf
+        sslmod_args = ["ssl_module", "modules/mod_somethingelse.so"]
+        rootconf_ifmod = self.config.parser.get_ifmod(
+            parser.get_aug_path(self.config.parser.loc["default"]),
+            "!mod_ssl.c", beginning=True)
+        self.config.parser.add_dir(rootconf_ifmod[:-1], "LoadModule", sslmod_args)
+        self.config.save()
+        self.config.assoc["test.example.com"] = self.vh_truth[0]
+        pre_matches = self.config.parser.find_dir("LoadModule",
+                                                  "ssl_module", exclude=False)
+        with mock.patch("certbot_apache.override_centos.logger.info") as mock_log:
+            self.config.deploy_cert(
+                "random.demo", "example/cert.pem", "example/key.pem",
+                "example/cert_chain.pem", "example/fullchain.pem")
+            self.assertTrue(mock_log.called)
+            self.assertTrue("Multiple different LoadModule" in mock_log.call_args[0][0])
+
+        post_matches = self.config.parser.find_dir("LoadModule",
+                                                   "ssl_module", exclude=False)
+        # Make sure that none was changed
+        self.assertEqual(pre_matches, post_matches)
+
+    def test_loadmod_not_found(self):
+        # Remove all existing LoadModule ssl_module... directives
+        orig_loadmods = self.config.parser.find_dir("LoadModule",
+                                                    "ssl_module",
+                                                    exclude=False)
+        for mod in orig_loadmods:
+            noarg_path = mod.rpartition("/")[0]
+            self.config.aug.remove(noarg_path)
+        self.config.save()
+        # get_all_args() is called for each LoadModule that was found
+        getall = "certbot_apache.override_centos.CentOSParser.get_all_args"
+        with mock.patch(getall) as mock_getall:
```

Comment:
nit: Instead of or in addition to mocking out this function, could we test that we didn't add the directives to the main configuration file?